modules/openapi-generator/src/main/resources/typescript-nestjs/api.module.mustache

```diff
@@ -0,0 +1,40 @@
+import { DynamicModule, HttpService, HttpModule, Module, Global } from '@nestjs/common';
+import { Configuration } from './configuration';
+import { BASE_PATH } from './variables';
+
+{{#apiInfo}}
+{{#apis}}
+import { {{classname}} } from './{{importPath}}';
+{{/apis}}
+{{/apiInfo}}
+
+@Global
+@Module({
+  imports:      [ HttpModule ],
+  exports:      [   
+    {{#apiInfo}}{{#apis}}{{classname}}{{#hasMore}},
+    {{/hasMore}}{{/apis}}{{/apiInfo}} 
+  ],
+  providers: [
+    {
+        provide: 'BASE_PATH',
+        useValue: BASE_PATH
+    },
+    {{#apiInfo}}{{#apis}}{{classname}}{{#hasMore}},
+    {{/hasMore}}{{/apis}}{{/apiInfo}} 
+  ]
+})
+export class ApiModule {
+    public static forRoot(configurationFactory: () => Configuration): DynamicModule {
+        return {
+            module: ApiModule,
+            providers: [ { provide: Configuration, useFactory: configurationFactory } ]
+        };
+    }
+
+    constructor( httpService: HttpService) {
+        if (!httpService) {
+            throw new Error('You need to import the {HttpModule in your AppModule!');
```

Comment:
Why is this check needed, if the `ApiModule` explicitly imports `HttpModule`?

Reply:
fixed, should not be here